src/wh_server_keystore.c

```diff
@@ -241,24 +246,21 @@ int wh_Server_HandleKeyRequest(whServerContext* server,
             XMEMCPY(meta->label, packet->keyCacheReq.label,
                 packet->keyCacheReq.labelSz);
         }
-        /* get a new id if one wasn't provided */
-        if (meta->id == WOLFHSM_ID_ERASED) {
-            ret =  hsmGetUniqueId(server);
-            if (ret > 0) {
-                meta->id = ret;
-                ret = 0;
-            }
-        }
+        /* get a new id if one wasn't provided, user has already been applied */
+        if (packet->keyCacheReq.id == WOLFHSM_KEYID_ERASED)
+            ret = hsmGetUniqueId(server, &meta->id);
         /* write the key */
         if (ret == 0)
             ret = hsmCacheKey(server, meta, in);
         if (ret == 0) {
-            packet->keyCacheRes.id = meta->id;
+            /* remove the user, client may set type */
+            packet->keyCacheRes.id = (meta->id & (~WOLFHSM_KEYUSER_MASK));
```

Comment:
Should we remove type as well, given that it isn't set by the client? From the client perspective it would be weird to get back an ID that is different than the one you sent, in case a client were to use that ID for bookkeeping reasons.